src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -334,6 +334,24 @@ public static Thread CurrentThread
             }
         }
 
+        private static int VerifyTimeoutMilliseconds(int millisecondsTimeout)
+        {
+            if (millisecondsTimeout < -1)
```

Comment:
Nit: I know you have copied this from CoreRT, but I am not a big fan of factoring out one liner argument validations like this just so that it can be use in a few places. The overhead of the extra method tends to be more than the savings.